src/VecSim/spaces/IP/IP_SVE_INT8.h

```diff
@@ -0,0 +1,101 @@
+/*
+ *Copyright Redis Ltd. 2021 - present
+ *Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ *the Server Side Public License v1 (SSPLv1).
+ */
+#pragma once
+#include "VecSim/spaces/space_includes.h"
+#include <arm_sve.h>
+
+static void InnerProductStep(const int8_t *&pVect1, const int8_t *&pVect2, size_t &offset,
+                             svfloat32_t &sum) {
+    svbool_t pg = svptrue_b8();
+
+    // Load int8 vectors
+    svint8_t v1_i8 = svld1_s8(pg, pVect1 + offset);
+    svint8_t v2_i8 = svld1_s8(pg, pVect2 + offset);
+
+    svfloat32_t ipf32 = svcvt_f32_s32_x(pg, svdot_s32(svdup_s32(0), v1_i8, v2_i8));
```

Comment:
why not accumulate the results with the expected `s32` type, and cast it to float only at the end?